eventsources/sources/bitbucketserver/start.go

```diff
@@ -93,6 +98,40 @@ func (router *Router) HandleRoute(writer http.ResponseWriter, request *http.Requ
 		return
 	}
 
+	// When SkipBranchRefsChangedOnOpenPR is enabled and webhook event type is repo:refs_changed,
```

Comment:
Why is there so specific logic hardcoded?

If there's a requirement like you mentioned in the PR description, instead of doing this, you can use filter feature in the Sensor take the events you are interested.

Reply:
I met with Pipekit folks and because of the API call required it does not seem possible. With auth and an adding a self-signed ca certificate. This addition goes a long way for Argo to replace the multibranch pipeline in Jenkins. If you want I can do this in a separate supporting/pleading my case a little better.